### DISCUSSION TITLE
How to hide product options

### DISCUSSION THREAD

**Discussion by addison74** [OpenMage/magento-lts#2809] [Questions & Answers]
I would like to mention I did not search the Internet for any resources yet. 

I am looking for a piece of code to hide custom options for a simple product. For example, a product has a custom option named "Color" with 3 rows (Red, Green, Blue). At a moment Green and Blue become temporary unavailable and the store admin doesn't want them listed. As OpenMage is built in the only way is to delete those two rows and put them back when they become available. I would like to skip this inefficient time consuming solution and add a checkbox named "Exclude" for every row. If it is checked that option should not be listed in Frontend. It is exactly the same behavior like when we exclude product images.

Please let me know if anyone has a solution for this issue. I think this should be implemented in OpenMage.

**Comment by kiatng** [comment]
That's an interesting feature. In my case, there are select elements where the options are dependent on dynamic factors and other selected options. I use js to hide/show the options.

**Reply by addison74** [reply]
**Advanced Product Options** produced by Mageworx has a lot of features, including conditional options. I evaluated this product since version 4.17.11 and discovered a lot of issues that I reported to the developer. At this moment there is no extension available to allow temporary exclusion for a custom option. This is an important feature in some cases, when the admin decides to hide for a while an option until it becomes available again.

This could be achieved by creating a Configurable Product with Simple Products. It works when the number is small but if the number is high it could create a performance issue on the server. I opened a discussion for this in this repository. In my case there are around 400 products with an average of 4 custom options. Making an evaluation we get this 400 Configurable Products + 400 x 4 Simple Products. In total there are 2000 Products against 400 Simple Products which use custom options only. For sure it will be a performance issues especially in the time load of a page when using Configurable Products.

**Reply by addison74** [reply]
For those interested in conditional custom options (children) Aitoc is offering a very good extension https://www.aitoc.com/magentomods-custom-options-templates.html. If swatches are requested I recommend Mageworx which has also the children feature. Anyway exclude a row remains a missing feature in both extensions.

**Comment by addison74** [comment]
I analyzed in more detail this feature that we could add to OpenMage to give flexibility to the custom options when not all are available at a moment. Instead of a checkbox dedicated to hiding an option, the most suitable would be a dropdown called Status with 3 variants Show / Disable / Hide. By default the first one is selected. The second one selected will make the option visible in the Frontend but it cannot be used. The third one will temporary remove the option from the Frontend.

**Comment by kiatng** [comment]
I have implemented custom options which are:
1. inputs of type `hidden`
2. `readonly` inputs based on some conditions
3. js to remove optional inputs entirely based on some conditions
4. select input with custom source (an example, country dropdown)
5. hide or show select options based on some conditions
6. add hint/note to options to aid users
